tests/functional/aws-node-sdk/test/bucket/aclUsingPredefinedGroups.js

```diff
@@ -0,0 +1,634 @@
+                        assert(er.code, errors.AccessDenied);
+                        done();
+                    };
+                    if (auth) {
+                        otherAccountBucketUtil.s3.listObjects(param, cb);
+                    } else {
+                        unauthenticatedRequest('listObjects', param, cb);
+                    }
+                });
+            });
+
+            it('should grant write access', done => {
+                s3.putBucketAcl({
+                    Bucket: testBucket,
+                    GrantWrite: grantUri,
+                }, err => {
+                    assert.ifError(err);
+                    const param = {
+                        Bucket: testBucket,
+                        Body: testBody,
+                        Key: testKey,
+                    };
+                    const cb = er => {
+                        assert.ifError(er);
+                        done();
+                    };
+                    if (auth) {
+                        otherAccountBucketUtil.s3.putObject(param, cb);
+                    } else {
+                        unauthenticatedRequest('putObject', param, cb);
+                    }
+                });
+            });
+
+            it('should grant write access with grant-full-control', done => {
+                s3.putBucketAcl({
+                    Bucket: testBucket,
+                    GrantFullControl: grantUri,
+                }, err => {
+                    assert.ifError(err);
+                    const param = {
+                        Bucket: testBucket,
+                        Body: testBody,
+                        Key: testKey,
+                    };
+                    const cb = er => {
+                        assert.ifError(er);
+                        done();
+                    };
+                    if (auth) {
+                        otherAccountBucketUtil.s3.putObject(param, cb);
+                    } else {
+                        unauthenticatedRequest('putObject', param, cb);
+                    }
+                });
+            });
+
+            it('should not grant write access', done => {
+                const aclParam = {
+                    Bucket: testBucket,
+                };
+                if (auth) {
+                    Object.assign(aclParam, {
+                        ACL: 'private',
+                    });
+                } else {
+                    Object.assign(aclParam, {
+                        GrantWrite: `uri=${constants.allAuthedUsersId}`,
+                    });
+                }
+                s3.putBucketAcl(aclParam, err => {
+                    assert.ifError(err);
+                    const param = {
+                        Bucket: testBucket,
+                        Body: testBody,
+                        Key: testKey,
+                    };
+                    const cb = er => {
+                        assert(er.code, errors.AccessDenied);
```

Comment:
Did you mean to use `assert.strictEqual`? This will return true if er.code is not 0, then use access denied as the message.

Reply:
Yep, it was a copy-paste.